### PR TITLE
Create raw request object to store and work with request level information

### PR DESCRIPTION
**Pull request by JacobBovee**
This adds a `RawRequest` object. Rather than passing around line and index we are storing them in this object which will be used for request level data, and data received prior to parsing our line. This fits better in our data model, it gives us a nice default object piggy backing off of our Select overload. Going forward the pattern when working with information received prior to creating our `IFrameLine` is the data should be stored in `RawRequest` and passed down as such. This is also currently a dependency of any pre-filters that will be built. 

Haven't given this too much thought yet but if we wanted to decouple pre-filters from `RawRequest` so it's up to the implemented class to define what, if any, data it needs we could do something like register RawRequst as a scoped item and update the values in our select overload. I don't like the idea of having that scoped state and letting those values get mutated at any point in our lifecycle though because it takes a lot of the trust away. Something that might be better is registering a list in RawRequests that we can push up to as we receive our lines `_lineBreaker.BreakIntoLines(input).Select((value, i) => RawRequest.PushLine(value, i)`. At that point we know where the most recent raw request is if we need to access that and a nice side affect is we'd be able to do things like look behinds on previous lines if need be. 

In that case our IPreFilter would be identical to IFilter so maybe even just add a "type" property for the inheriting class to implement that takes a FilterType (or FilterLifecycle maybe) enum of `Pre` or `Default`. Then it can be left up to a filter runner function to take in the LifeCycle value and run the relevant filters as needed. This isn't super necessary however I thought it could lead to a pattern where we have a single Filter base class that is inherited by all of our filters where we can keep shared logic, such as a static filter runner, and the process for adding additional LifeCycles for a filter is just a matter of adding to the enum rather than creating a new interface and registering it.

### CODE REVIEW

File: src/NetStackBeautifier.Core/Models/RawRequest.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+namespace NetStackBeautifier.Services
+{
+    public class RawRequest
+    {
+        public RawRequest(string value, int index)
+        {
+            Value = value;
+            Index = index;
+        }
+        public int Index { get; private set; }
```
Comment (xiaomi7732) [inline src/NetStackBeautifier.Core/Models/RawRequest.cs:10]: If there is no need to mutation for the values post ctor, there is no need for the private setter. For example:

```
        public int Index { get; }
```

File: src/NetStackBeautifier.Services/Filters/RetainedFilter.cs [hunk 2]
```diff
@@ -3,13 +3,13 @@ namespace NetStackBeautifier.Services.Filters;
 internal class RetainedFilter<T> : IPreFilter<T>
     where T : IBeautifier
 {
-    public void Filter(IFrameLine frameLine, string fullLine, int startingIndex)
+    public void Filter(IFrameLine frameLine, RawRequest rawRequest)
     {
-        if (frameLine == null)
+        if (frameLine == null || rawRequest == null)
```
Comment (xiaomi7732) [inline src/NetStackBeautifier.Services/Filters/RetainedFilter.cs:8]: Only personal opinion: `==` in C# is overwritable. Use `is` or `is not` than `==`, for example:

```csharp
if (frameLine is null || rawRequest is null)
```
But I am not too strong against what you have there.

**Review by xiaomi7732 — APPROVED** [review]
Only a few minor points. It looks good to me.

File: src/NetStackBeautifier.Core/Models/RawRequest.cs (hunk 1 above)
Comment (JacobBovee) [reply src/NetStackBeautifier.Core/Models/RawRequest.cs]: Sounds good, fixed.

File: src/NetStackBeautifier.Services/Filters/RetainedFilter.cs (hunk 2 above)
Comment (JacobBovee) [reply src/NetStackBeautifier.Services/Filters/RetainedFilter.cs]: Agreed, fixed.

**Review by xiaomi7732 — APPROVED** [review]
LGTM.

**Comment by xiaomi7732** [comment]
Tag #6

**Pull request merged**